File: CS50X/pset6/readability/readability.py
from cs50 import get_string

text = get_string("TEXT: ")


def main():
    L = letters(text)
    W = words(text)
    S = sentences(text)
    print(L)
    print(W)
    print(S)
    L = (100/W)*L
    S = (100/W)*S
    
    print(L)
    print(S)
    
    index = round(0.0588 * L - 0.296 * S - 15.8)
    
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def letters(text):
    counter = 0
    text = text.lower()
    for i in range(len(text)):
        if ord(text[i]) >= 97 and ord(text[i]) <= 122:
            counter += 1
            
    return counter
    
    
def words(text):
    counter = 1
    for i in range(len(text)):
        if ord(text[i]) == 32:
            counter += 1
    return counter
    
    
def sentences(text):
    counter = 0
    for i in range(len(text)):
        if ord(text[i]) == 33 or ord(text[i]) == 46 or ord(text[i]) == 63:
            counter += 1
    return counter
    
    
main()

File: CS50X/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string input);
int count_words(string input);
int count_sen(string input);


int main (void){
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sen = count_sen(text);
    printf("%i\n", letters);
    printf("%i\n", words);
    printf("%i\n", sen);
    
    float L = (100/(float)words) * letters;
    float S = (100/(float)words) * sen;
    printf("%f\n", L);
    printf("%f\n", S);
    
    int grade = round(0.0588 * L - 0.296 * S - 15.8);
    if(grade<1){
        printf("Before Grade 1\n");
    }
    else if(grade>=16){
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %i\n", grade);
    }
    
}



int count_letters(string input){
    int m = 0;
    for (int i = 0,n =strlen(input); i<n; i++){
        if((int) toupper(input[i]) >=65 && (int) toupper(input[i]) <=95){
        m++;
        }
    }
    return m;
}

int count_words(string input){
    int word = 1;
    for(int i = 0, n = strlen(input); i<n; i++){
        if((int) toupper(input[i]) == 32){
            word++;
        }
    }
    return word;
}

int count_sen(string input){
    int sen = 0;
    for(int i = 0,n =strlen(input); i<n; i++){
        if((int) toupper(input[i]) == 33 || (int) toupper(input[i]) == 46 || (int) toupper(input[i]) == 63){
            sen++;
        }
    }
    return sen;
}